phantom_token.c

```diff
@@ -651,7 +542,48 @@ static ngx_int_t introspection_response_handler(ngx_http_request_t *request, voi
     }
     else
     {
-        jwt_start = request->header_end + sizeof("\r\n") - 1; // FIXME: Won't work if JWT is large
+        read_response = true;
+    }
+    
+#else
+    read_response = true;
+#endif
+
+    jwt_len = request->headers_out.content_length_n;
+    bearer_jwt_len = BEARER_SIZE + jwt_len;
+
+    // When caching is not used, the introspection response is read directly.
+    // This method only receives a single response buffer per introspection request, which needs to contain the full JWT.
+    if (read_response)
+    {
+        // We use proxy_pass to call the introspection endpoint, so the ngx_http_proxy_module provides the response to the subrequest.
+        // The response is returned as an upstream buffer that NGINX prepares when it calls ngx_http_parse_header_line.
+        // This sets request->header_end and points request->upstream->buffer.pos past headers to the body content.
+        // - https://github.com/nginx/nginx/blob/master/src/http/modules/ngx_http_proxy_module.c#L1905
+        // - https://github.com/nginx/nginx/blob/master/src/http/ngx_http_parse.c#L816
+        jwt_start = request->header_end + sizeof("\r\n") - 1;
```

Comment:
I guess this could be based on `request->upstream->buffer.pos` also? Would be cleaner. I understand if you don't want to do more changes.

Reply:
OK - will update to that - I agree it is cleaner and we already take a dependency on this NGINX behavior